### PR TITLE
format StmtTryStar

### PR DESCRIPTION
**Pull request by davidszotten**
refactor StmtTry to share most of impl with StmtTryStar

### CODE REVIEW

File: crates/ruff_python_formatter/src/other/except_handler_except_handler.rs [hunk 1]
```diff
@@ -2,12 +2,26 @@ use crate::comments::trailing_comments;
...
 use crate::{FormatNodeRule, PyFormatter};
+use ruff_formatter::FormatRuleWithOptions;
 use ruff_formatter::{write, Buffer, FormatResult};
 use ruff_python_ast::node::AstNode;
 use rustpython_parser::ast::ExceptHandlerExceptHandler;
 
 #[derive(Default)]
-pub struct FormatExceptHandlerExceptHandler;
+pub struct FormatExceptHandlerExceptHandler {
+    has_star: bool,
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/other/except_handler_except_handler.rs:12]: Nit: I think an enum `Starred` with the options `Yes` and `No` or `ExceptHandlerKind` with the variants `Starred` and `Regular` would improve readability on the `with_options` call site

File: crates/ruff_python_formatter/src/other/except_handler_except_handler.rs [hunk 2]
```diff
@@ -25,7 +39,7 @@ impl FormatNodeRule<ExceptHandlerExceptHandler> for FormatExceptHandlerExceptHan
         let comments_info = f.context().comments().clone();
         let dangling_comments = comments_info.dangling_comments(item.as_any_node_ref());
 
-        write!(f, [text("except")])?;
+        write!(f, [text("except"), self.has_star.then(|| text("*"))])?;
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/other/except_handler_except_handler.rs:42]: Nit
```suggestion
        write!(f, [text("except"), self.has_star.then_some(text("*"))])?;
```

Calling `text` is cheap (it compiles to a const)

**Review by MichaReiser — APPROVED** [review]
Nice! I've two small nits. Let me know if you want to address them or if I should merge as is.

File: crates/ruff_python_formatter/src/other/except_handler_except_handler.rs (hunk 2 above)
Comment (davidszotten) [reply crates/ruff_python_formatter/src/other/except_handler_except_handler.rs:42]: wasn't saving the call, just missed that method existing 😆

**Pull request merged**